Controllers/FileReader.cs

```diff
@@ -0,0 +1,177 @@
+            }
+        }
+
+        private void ReadSEB(StreamReader reader, bool quotes)
+        {
+            string line = reader.ReadLine();
+            while (line != null)
+            {
+                line = reader.ReadLine();
+                if (line != null)
+                {
+                    string pattern = @"\s*;\s*";
+
+                    string[] tokens = System.Text.RegularExpressions.Regex.Split(
+                        line, pattern);
+
+                    decimal d = decimal.Parse(tokens[3].Replace(',', '.'));
+                    if (quotes)
+                    {
+                        if (tokens[14] == "\"D\"")
+                            d = d * -1;
+                        string details = tokens[9].Substring(1, tokens[9].Length - 2);
+                        string counterParty = tokens[4].Substring(1, tokens[4].Length - 2);
+
+                        AddTransaction(DateTime.Parse(tokens[1]), d, counterParty, details);
+                    }
+                    else
+                    {
+                        if (tokens[14] == "D")
+                            d = d * -1;
+                        string details = tokens[9];
+                        string counterParty = tokens[4];
+
+                        AddTransaction(DateTime.Parse(tokens[1]), d, counterParty, details);
+                    }
+
+                }
+            }
+        }
+        private void ReadLum(StreamReader reader)
+        {
+            string line = reader.ReadLine();
+            while (line != null)
+            {
+                line = reader.ReadLine();
+                if (line != null)
+                {
+                    string pattern = @"\s*;\s*";
+
+                    string[] tokens = System.Text.RegularExpressions.Regex.Split(
+                        line, pattern);
+
+                    int y = Int32.Parse(tokens[1].Substring(0, 4));
+                    int m = Int32.Parse(tokens[1].Substring(4, 2));
+                    int day = Int32.Parse(tokens[1].Substring(6, 2));
+                    int h = Int32.Parse(tokens[2].Substring(0, 2));
+                    int min = Int32.Parse(tokens[2].Substring(2, 2));
+                    int s = Int32.Parse(tokens[2].Substring(4, 2));
+
+                    decimal d = decimal.Parse(tokens[3]);
+                    if (tokens[5] == "D")
+                        d = d * -1;
+                    AddTransaction(new DateTime(y, m, day, h, min, s), d, tokens[12], tokens[16]);
+                    
+                }
+            }
+        }
+
+        private void AddTransaction(DateTime time, decimal amount, string counterParty, string details)
+        {
+            Database db = new Database();
+            Transaction newTransaction = new Transaction();
+            newTransaction.TrTime = time;
+            newTransaction.Amount = amount;
+            newTransaction.CounterParty = counterParty;
+            newTransaction.Details = details;
```

Comment:
Can be written as:
`
Goal newGoal = new Goal
                {
                    Deadlinedate = date,
                    Amount = amountInDecimal,
                    Title = name,
                    Description = description
                };
`

Reply:
Thanks